### PR TITLE
Add tabs component

### PR DESCRIPTION
**Pull request by JuliaHempel**
Add a tabs component

Todo:
- [x] Write descriptions
- [x] Add styling
- [x] Make buttons dropdown on small screens if 3 or more tabs
- [x] Add tests
- [x] Make sure it's not called index when shown in menu to the left
- [x] Make sure it's accessible

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -58,6 +58,7 @@ export * from './components/organisms/Form';
 export * from './components/organisms/Accordion';
 export { default as Navbar, navbarLinkStyles } from './components/organisms/Navbar/';
 export { default as Card } from './components/organisms/Card';
+export * from './components/organisms/Tabs';
```
Comment (JuliaHempel) [inline src/index.ts:61]: I have no idea where to add it in order for it to actually show up.

Comment (poteirard) [reply src/index.ts:61]: If it's an organism or template you have to manually add the section in `/styleguide.config` follow the same pattern as the rest of the organisms.

File: src/components/organisms/Tabs/TabButtons/TabButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  useEffect(() => {
+    defaultTab && setActiveTab(defaultTab);
+  }, []);
+
+  return (
+    <>
+      {tabButtons.length > 2 && width <= grid.breakpoints.s ? (
+        <Dropdown
+          defaultValue={defaultTab}
+          onChange={(event: ChangeEvent<HTMLSelectElement>) => setActiveTab(event.target.value)}
```
Comment (poteirard) [inline src/components/organisms/Tabs/TabButtons/TabButtons.tsx:28]: Please add the handler in an `useCallback` hook. More info [here](https://medium.com/@abdullahdanyal/how-to-improve-performance-issues-caused-by-arrow-functions-in-react-11dd25527ddf)

File: src/components/organisms/Tabs/TabButtons/TabButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        >
+          {tabButtons.map((tabButton: TabButtonProps) => (
```
Comment (poteirard) [inline src/components/organisms/Tabs/TabButtons/TabButtons.tsx:30]: this `: TabButtonProps` shouldn't be necessary as it's already inferred by the compiler when you declared `tabButtons: TabButtonProps[];` 

Same in line 37

File: src/components/organisms/Tabs/TabButton/TabButton.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      }
+    `}
+`;
+
+const TabButton = ({ title, tabId }: TabButtonProps) => {
+  const { activeTab, getTabButtonHTMLProps } = useTabsContext();
+  const tabProps = getTabButtonHTMLProps(tabId);
+
+  return (
+    <StyledButton activetab={activeTab} tabid={tabId} styling="link" {...tabProps}>
```
Comment (JuliaHempel) [inline src/components/organisms/Tabs/TabButton/TabButton.tsx:50]: Please note the styling of the buttons doesn't match the designs due to the design the buttons have in sketch is not a design we currently have for the buttons. Lianne is on it.

Comment (admmasters) [inline src/components/organisms/Tabs/TabButton/TabButton.tsx:50]: We should probably have consistent casing of the prop names: e.g. activeTab vs activetab and tabid vs tabId etc

File: src/components/organisms/Tabs/TabButtons/TabButtons.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+### Summary
+
+`<Tabs.Buttons />` is supposed to be a child of `<Tabs />`. It renders the tab buttons.
```
Comment (admmasters) [inline src/components/organisms/Tabs/TabButtons/TabButtons.md:3]: is supposed to -> should

File: src/components/organisms/Tabs/TabButtons/TabButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import Dropdown from '../../../atoms/Dropdown/Dropdown';
+import { grid, colors } from '../../../../constants';
+import TabButton, { TabButtonProps } from '../TabButton/TabButton';
+import { DropdownOption } from '../../../..';
+import styled from 'styled-components';
+
+interface TabButtonsProps {
+  tabButtons: TabButtonProps[];
+  defaultTab?: string;
+  dataAutomation?: string;
```
Comment (admmasters) [inline src/components/organisms/Tabs/TabButtons/TabButtons.tsx:13]: not necessarily for this PR - not sure what other people think but I'm actually thinking its better to just have the actual prop as `["data-automation"]` for consistency with the underlying element

File: src/components/organisms/Tabs/TabButton/TabButton.tsx (hunk 4 above)
Comment (JuliaHempel) [reply src/components/organisms/Tabs/TabButton/TabButton.tsx:51]: I had it like that but the cypress tests complained about it

Comment (JuliaHempel) [reply src/components/organisms/Tabs/TabButton/TabButton.tsx:51]: I get the [unknown prop error](https://reactjs.org/warnings/unknown-prop.html) when I use the camel caseing. Please let me know how I get rid of it.

Comment (JuliaHempel) [reply src/components/organisms/Tabs/TabButton/TabButton.tsx:50]: Lianne has approved the buttons as they are now.

**Review by admmasters — APPROVED** [review]

**Pull request merged**